python/taichi/lang/struct.py

```diff
@@ -293,60 +293,60 @@ class StructField(Field):
     def __init__(self, field_dict, name=None):
         # will not call Field initializer
         self.field_dict = field_dict
-        self._name = name
-        self.register_fields()
+        self.__name = name
```

Comment:
For this we can directly say `self.name = name` and remove the property below.

Reply:
done